src/main/java/com/mixpanel/android/viewcrawler/ConnectivityIndicator.java

```diff
@@ -0,0 +1,201 @@
+        }
+        return sInstance;
+    }
+
+    public static void stop()
+    {
+        sInstance = null;
+    }
+
+    private View.OnTouchListener getConnectivityIndicatorTouchListener() {
+        return new View.OnTouchListener()
+        {
+            private float initialTouchX;
+            private float initialTouchY;
+            private boolean isDragEvent=false;
+
+            public boolean onTouch(View v,MotionEvent event)
+            {
+                if((event == null) || (ConnectivityIndicator.this.params == null) || (ConnectivityIndicator.this.windowManager == null)){
+                    return false;
+                }
+                System.out.println(event.getAction() + " , " + event.getRawX() + " , " + event.getRawY());
+                switch(event.getAction())
+                {
+                    case MotionEvent.ACTION_DOWN:
+                        this.initialTouchX = event.getRawX();
+                        this.initialTouchY = event.getRawY();
+                        this.isDragEvent = false;
+                        return true;
+                    case MotionEvent.ACTION_UP:
+                        if(this.isDragEvent)
+                        {
+                            ConnectivityIndicator.this.currentX = ConnectivityIndicator.this.params.x;
+                            ConnectivityIndicator.this.currentY = ConnectivityIndicator.this.params.y;
+                        }
+                        return true;
+                    case MotionEvent.ACTION_MOVE:
+                        float diffX = event.getRawX() - this.initialTouchX;
+                        float diffY = event.getRawY() - this.initialTouchY;
+                        ConnectivityIndicator.this.params.x = (ConnectivityIndicator.this.currentX + (int)diffX);
+                        ConnectivityIndicator.this.params.y = (ConnectivityIndicator.this.currentY + (int)diffY);
+                        ConnectivityIndicator.this.windowManager.updateViewLayout(ConnectivityIndicator.this, ConnectivityIndicator.this.params);
+                        int slop = ViewConfiguration.get(v.getContext()).getScaledTouchSlop();
+                        if((diffX > slop) || (diffY > slop)){
+                            this.isDragEvent = true;
+                        }
+                        return true;
+                }
+                return false;
+            }
+        };
+    }
+
+    private ConnectivityIndicator(Context context)
+    {
+        super(context);
+        setOnTouchListener(getConnectivityIndicatorTouchListener());
+        ImageView iconView = new ImageView(context);
+        iconView.setImageResource(R.drawable.com_mixpanel_android_connectivityabtest);
+        addView(iconView);
+    }
+
+    public static void show(final Activity activity)
+    {
+        if ((sInstance == null) || (!ViewCrawler.mMixpanelInEditMode) || activity == null) {
+            return;
+        }
+        if (sInstance.isShown) {
```

Comment:
No need to have this variable. If this view has no parent it means is not shown :)